### DISCUSSION TITLE
Borg Backup encountered error....

### DISCUSSION THREAD

**Discussion by sghorshine** [borgbackup/borg#7971] [Q&A] [answered]
Backup started for all databases at Fri 22 Dec 2023 12:02:01 AM +0545
Export of individual DB started
Starting backup of Office Server at 20231222
Creating archive at "mocobackup@localhost:/home/mocobackup/repos/office-server::20231222-office-server"
Data integrity error: Segment entry checksum mismatch [segment 3545, offset 33520440]
Traceback (most recent call last):
  File "borg/archiver.py", line 4621, in main
  File "borg/archiver.py", line 4553, in run
  File "borg/archiver.py", line 177, in wrapper
  File "borg/archiver.py", line 590, in do_create
  File "borg/archiver.py", line 555, in create_inner
  File "borg/archive.py", line 527, in save
  File "borg/remote.py", line 477, in do_rpc
  File "borg/remote.py", line 712, in call
  File "borg/remote.py", line 780, in call_many
  File "borg/remote.py", line 740, in handle_error
borg.helpers.IntegrityError: Data integrity error: Segment entry checksum mismatch [segment 3545, offset 33520440]

Platform: Linux office 5.4.0-167-generic #184-Ubuntu SMP Tue Oct 31 09:21:49 UTC 2023 x86_64
Linux: debian bullseye/sid
Borg: 1.1.15  Python: CPython 3.7.9 msgpack: 0.5.6
PID: 286188  CWD: /root
sys.argv: ['borg', 'create', 'mocobackup@localhost:/home/mocobackup/repos/office-server::20231222-office-server', '/var/www', '/etc/apache2/sites-available', '/etc/mysql', '/var/lib/gitea', '/etc/freeradius/3.0', '/veracrypt_recover_disk', '/root/DB', '/home/ams', '/home/ma', '-v', '--stats', '--compression', 'lzma']
SSH_ORIGINAL_COMMAND: None

Backup finsihed with an error at 2023-12-22 00:02:23
Backup started for all databases at Fri 22 Dec 2023 09:47:45 AM +0545
Export of individual DB started
Starting backup of Office Server at 20231222
Creating archive at "mocobackup@localhost:/home/mocobackup/repos/office-server::20231222-office-server"
Synchronizing chunks cache...
Archives: 665, w/ cached Idx: 30, w/ outdated Idx: 0, w/o cached Idx: 635.
Reading cached archive chunk index for 20220223-office-server ...
Merging into master chunks index ...
...
Fetching and building archive index for 20231222-office-server ...
Merging into master chunks index ...
Done.
Archive 20231222-office-server already exists

Backup finsihed with an error at 2023-12-22 09:49:44
Backup started for all databases at Fri 22 Dec 2023 10:08:47 AM +0545
Export of individual DB started
Starting backup of Office Server at 20231222
Creating archive at "mocobackup@localhost:/home/mocobackup/repos/office-server::20231222-office-server"
Synchronizing chunks cache...
Archives: 664, w/ cached Idx: 665, w/ outdated Idx: 1, w/o cached Idx: 0.
Reading cached archive chunk index for 20220223-office-server ...
Merging into master chunks index ...
...
Reading cached archive chunk index for 20231221-office-server ...
Merging into master chunks index ...
Done.
Data integrity error: Segment entry checksum mismatch [segment 3545, offset 33520440]
Traceback (most recent call last):
  File "borg/archiver.py", line 4621, in main
  File "borg/archiver.py", line 4553, in run
  File "borg/archiver.py", line 177, in wrapper
  File "borg/archiver.py", line 590, in do_create
  File "borg/archiver.py", line 555, in create_inner
  File "borg/archive.py", line 527, in save
  File "borg/remote.py", line 477, in do_rpc
  File "borg/remote.py", line 712, in call
  File "borg/remote.py", line 780, in call_many
  File "borg/remote.py", line 740, in handle_error
borg.helpers.IntegrityError: Data integrity error: Segment entry checksum mismatch [segment 3545, offset 33520440]

Platform: Linux office 5.4.0-167-generic #184-Ubuntu SMP Tue Oct 31 09:21:49 UTC 2023 x86_64
Linux: debian bullseye/sid
Borg: 1.1.15  Python: CPython 3.7.9 msgpack: 0.5.6
PID: 291401  CWD: /usr/local/sbin
sys.argv: ['borg', 'create', 'mocobackup@localhost:/home/mocobackup/repos/office-server::20231222-office-server', '/var/www', '/etc/apache2/sites-available', '/etc/mysql', '/var/lib/gitea', '/etc/freeradius/3.0', '/veracrypt_recover_disk', '/root/DB', '/home/ams', '/home/ma', '-v', '--stats', '--compression', 'lzma']
SSH_ORIGINAL_COMMAND: None

Backup finsihed with an error at 2023-12-22 10:09:10 /var/log/borg/office/20231222.log

**Comment by infectormp** [comment]
https://borgbackup.readthedocs.io/en/stable/usage/check.html#description
Could you also update to latest borg version?

**Reply by sghorshine** [reply]
Can you please how I update Borg backup to latest version? 

NAME="Ubuntu"
VERSION="20.04.6 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.6 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal

**Reply by sghorshine** [reply]
It would be nice if I could get my hands on documentation on how to update borg to latest version? @infectormp

**Comment by ThomasWaldmann** [comment]
@witten Does that typo still need fixing in borgmatic?:
```
Backup finsihed with an error at 2023-12-22 10:09:10 /var/log/borg/office/20231222.log
```

**Reply by sghorshine** [reply]
Is this typo from the installation itself I was always wondering why this typo even though the script to create borg repo is all correctly spelled.

`Backup finsihed with an error at 2023-12-22 10:09:10 /var/log/borg/office/20231222.log`

**Reply by ThomasWaldmann** [reply]
I suspect that this message comes from borgmatic, that's why I pinged @witten.

**Reply by witten** [reply]
Thanks for checking, but that message isn't from borgmatic and the other messages don't look like borgmatic logs. Maybe this is a personal Borg backup wrapper script?

**Reply by ThomasWaldmann** [reply]
@witten Oops, sorry for the noise then. I saw "Backup started for databases..." and thought it is borgmatic.

@sghorshine guess must be in your script then.

**Comment by ThomasWaldmann** [comment]
@sghorshine It looks like one entry in a segment file in the repo fails the checksum check, do what @infectormp suggested (run `borg check --repair ...`).

**Reply by sghorshine** [reply]
Is there any consequences if I run the borg check --repair <repository_name>::<backup_file> ?

**Reply by ThomasWaldmann** [reply, marked as answer]
Well, the usual consequence is that your repo will be repaired. :-)
Please read the borg check documentation for the details and important hints.

For borg check, you must give only the repo, no `::archive_name`.
